ReactWindows/ReactNative.Net46/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -346,21 +348,18 @@ public void OnDestroy()
 
         private async Task<IFolder> GetAsyncStorageFolder(bool createIfNotExists)
         {
-            if (_cachedFolder == null)
+            var localFolder = FileSystem.Current.LocalStorage;
+            var existsCheck = await localFolder.CheckExistsAsync(AsyncStorageHelpers.DirectoryName).ConfigureAwait(false);
+            if (existsCheck == ExistenceCheckResult.FolderExists)
             {
-                var localFolder = FileSystem.Current.LocalStorage;
-
-                if (localFolder.CheckExistsAsync(AsyncStorageHelpers.DirectoryName).Result == ExistenceCheckResult.FolderExists)
```

Comment:
In UWP everything is correct. In contrast of _PCLStorage_, _Windows.Storage_ has method `TryGetItemAsync` which is used there.